wg-data-protection/charter.md

```diff
@@ -0,0 +1,81 @@
+# WG Data Protection Charter
+
+This charter adheres to the [wg-governance] guidance as well as
+the general conventions described in the [Kubernetes Charter README] and
+the Roles and Organization Management outlined in [sig-governance], where
+applicable to a Working Group.
+
+
+## Scope
+
+Data protection in Kubernetes context typically involves backup and recovery
+of two types of entities:
+* Kubernetes API object resources
+* Persistent volume data
+We consider it a complicated and layered problem, including backup and recovery
+at persistent volume level, application level, and cluster level. Part of the
+working group’s charter is to define what Kubernetes native constructs are
+required to achieve these goals.
+
+The Data Protection Working Group is organized with the goal of providing
+a cross SIG forum to discuss how to support data protection in Kubernetes,
+identify missing functionality, and work together to design features that
+are needed to achieve the goal.
+
+### In scope
+
+* High level discussions on what it means to support data protection in Kubernetes at different levels and how to do it.
+* Design discussions on specific topics related to data protection and disaster recovery support.
+* Document results of discussions and investigations in a linkable medium.
+
+Potential design topics include, but are not limited to the following:
+* Read data from a snapshot without creating a new volume.
+* Volume backups
+* Data populator
+* Retrieve diffs between two snapshots (block and file level)
+* Consistency volume groups (group snapshot)
+* Application snapshot, backup, and recovery
```

Comment:
As recovery can usually take time based on amount of data/speed etc, worth to add application migration as well in the scope.. consider the usecase like application upgrades

Reply:
Good point. I agree that application migration may be involved during recovery.  I'm hesitant to add it explicitly in the charter though because "migration" itself has a very wide scope and it can be used in other non-recovery related cases as well.  Application upgrade is another big topic of its own.  

What we have listed here are high level potential topics.  As a WG, we can discuss in detail how to design and implement these features.